src/core/model.jl

```diff
@@ -262,7 +262,17 @@ datumdef(m::Model, comp_name::Symbol, item::Symbol) = datumdef(compdef(m.md, com
 Return the dimension names for the variable or parameter `datum_name`
 in the given component `comp_name` in model `m`.
 """
-dim_names(m::Model, comp_name::Symbol, datum_name::Symbol) = dim_names(compdef(m, comp_name), datum_name)
+function dim_names(m::Model, comp_name::Symbol, datum_name::Symbol)
+    try
+        return dim_names(compdef(m, comp_name), datum_name)
+    catch e
```

Comment:
I'm guessing that the error here occurs if `comp_def` isn't in the top component?  I'm thinking it would be better to have an if-then rather than a try-catch just for clarity. In fact perhaps we just take out the shortcut for now, the second loop should work for any comp_name.